### PR TITLE
Refactor Settings to new theme and update theme file

### PR DESCRIPTION
**Pull request by amanalili**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android
- [ ] n/a

### CODE REVIEW

File: src/components/common/SettingsHeaderRow.js [hunk 1]
```diff
@@ -42,12 +44,13 @@ const rawStyles = {
   text: {
     ...nightText('row-left'),
```
Comment (amanalili) [inline src/components/common/SettingsHeaderRow.js:45]: remove nightText

File: src/components/common/SettingsLabelRow.js [hunk 2]
```diff
@@ -1,32 +1,38 @@
...
+  const { disabled, icon, text, theme, right, onPress } = props
+  const styles = getStyles(theme)
 
   return <SettingsRow disabled={disabled} icon={icon} text={text} right={<Text style={styles.labelText}>{right}</Text>} onPress={onPress} />
 }
 
-const styles = {
+const getStyles = cacheStyles(theme => ({
   labelText: {
     ...dayText('bold', 'small'),
```
Comment (amanalili) [inline src/components/common/SettingsLabelRow.js:32]: remove dayText

File: src/components/common/SettingsLabelRow.js [hunk 3]
```diff
@@ -1,32 +1,38 @@
...
-    color: THEME.COLORS.SECONDARY
+    fontSize: theme.rem(1),
+    color: theme.settingsTextMintColor
```
Comment (amanalili) [inline src/components/common/SettingsLabelRow.js:34]: rename settingsTextMintColor

File: src/components/common/SettingsRow.js [hunk 4]
```diff
@@ -1,67 +1,75 @@
...
+  marginBottom: {
+    marginBottom: theme.rem(1 / 16)
+  },
+  text: {
+    ...dayText('row-left'),
+    color: theme.primaryText,
+    flexGrow: 1
+  },
+  disabledText: {
+    ...dayText('row-left'),
```
Comment (amanalili) [inline src/components/common/SettingsRow.js:63]: remove dayText

File: src/components/scenes/SettingsOverviewScene.js [hunk 5]
```diff
@@ -161,18 +165,18 @@ export default class SettingsOverview extends React.Component<Props, State> {
...
         <ScrollView>
           <SettingsHeaderRow
-            icon={<FontAwesomeIcon name="user-o" color={THEME.COLORS.WHITE} size={iconSize} />}
+            icon={<FontAwesomeIcon name="user-o" color={theme.settingsIconColor} size={iconSize} />}
             text={`${s.strings.settings_account_title_cap}: ${this.props.username}`}
           />
           <SettingsRow
             text={s.strings[this.props.lockButton]}
-            right={<IonIcon name={this.props.lockButtonIcon} color={THEME.COLORS.GRAY_1} size={iconSize} />}
+            right={<IonIcon name={this.props.lockButtonIcon} color={theme.settingsIconMintColor} size={iconSize} />}
```
Comment (amanalili) [inline src/components/scenes/SettingsOverviewScene.js:179]: rename settingsIconMintColor

File: src/components/scenes/SettingsOverviewScene.js [hunk 6]
```diff
@@ -250,17 +258,27 @@ export default class SettingsOverview extends React.Component<Props, State> {
...
+      resizeMode: 'contain'
+    },
+    bottomArea: {
+      padding: theme.rem(2)
+    },
+    button: {
+      width: '100%',
+      height: theme.rem(3),
+      borderRadius: theme.rem(1.5),
+      backgroundColor: theme.settingsButtonBackgroud
```
Comment (amanalili) [inline src/components/scenes/SettingsOverviewScene.js:276]: settingsButtonBackgroud = settingsButtonBackground

File: src/theme/variables/edgeDark.js [hunk 7]
```diff
@@ -51,5 +54,17 @@ export const edgeDark: Theme = {
   background2: pallete.darkBlueNavyGradient2,
   cancelButton: pallete.white,
   accentTextPositive: pallete.accentGreen,
-  accentTextNegative: pallete.accentRed
+  accentTextNegative: pallete.accentRed,
+  transparent: pallete.transparent,
+
+  // Settings theme:
+  settingsIconColor: pallete.white,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:85]: rename all to settingsRow

Comment (amanalili) [reply src/theme/variables/edgeDark.js:85]: remove color naming

File: src/theme/variables/edgeDark.js [hunk 8]
```diff
@@ -56,6 +56,12 @@ export const edgeDark: Theme = {
   accentTextPositive: pallete.accentGreen,
   accentTextNegative: pallete.accentRed,
   transparent: pallete.transparent,
+  keyboardTopViewBackgroundDefault: pallete.white,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:72]: add comment  that all the defaults go away when light mode

File: src/components/scenes/TransactionsExportScene.js [hunk 9]
```diff
@@ -361,4 +364,10 @@ const getStyles = cacheStyles(theme => ({
   }
 }))
 
-export const TransactionsExportSceneComponent = withTheme(TransactionsExportScene)
+export const TransactionsExportScene = connect((state: StateType, ownProps: OwnProps): StateProps => {
+  const denominationObject = getDisplayDenomination(state, ownProps.currencyCode)
+  const denomination = denominationObject.multiplier
+  return {
+    denomination
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:365]: rename to multiplier

File: src/components/scenes/CurrencySettingsScene.js [hunk 10]
```diff
@@ -112,8 +113,8 @@ export class CurrencySettingsComponent extends React.Component<Props, State> {
           {this.props.denominations.map(denomination => {
             const key = denomination.multiplier
             const left = (
-              <Text style={{ ...dayText('row-left'), color: THEME.COLORS.GRAY_1 }}>
-                <Text style={{ fontFamily: THEME.FONTS.SYMBOLS }}>{denomination.symbol}</Text> - {denomination.name}
+              <Text style={{ ...dayText('row-left'), color: theme.primaryText }}>
```
Comment (amanalili) [inline src/components/scenes/CurrencySettingsScene.js:116]: remove daytext

File: src/components/scenes/PromotionSettingsScene.js [hunk 11]
```diff
@@ -93,17 +95,17 @@ export class PromotionSettingsComponent extends React.Component<Props> {
...
-const space = THEME.rem(0.5)
-const rawStyles = {
+const getStyles = cacheStyles(theme => ({
   textBlock: {
-    padding: space
+    backgroundColor: theme.settingsRowBackground,
+    padding: theme.rem(0.5)
   },
   textRow: {
     ...dayText(),
```
Comment (amanalili) [inline src/components/scenes/PromotionSettingsScene.js:104]: remove dayText

File: src/theme/variables/edgeDark.js [hunk 12]
```diff
@@ -14,88 +14,167 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
-  selectButtonText: pallete.edgeMint,
-  modalBody: pallete.edgeNavy,
-  modalClose: pallete.white,
-  modalBackgroundShadow: pallete.black,
+  // Icons
+  icon: palette.white,
+  iconHighlighted: palette.edgeMint,
+  tileIcon: palette.white,
+  tileIconHighlighted: palette.edgeMint,
+  closeIcon: palette.gray,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:57]: Why is this different from modalCloseIcon

File: src/theme/variables/edgeDark.js [hunk 13]
```diff
@@ -14,88 +14,167 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
-  accentTextNegative: pallete.accentRed,
+  modalCloseIcon: palette.white,
+
+  // Tile
+  listHeaderBackground: palette.edgeNavy,
+  tileBackground: palette.edgeBlue,
+  tileBackgroundMuted: palette.mutedBlue, // change from mutedTileBackground
+
+  // Section
+  section: palette.edgeBlue,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:76]: rename section to settingsRow

File: src/components/common/SettingsHeaderRow.js [hunk 14]
```diff
@@ -33,7 +33,7 @@ const getStyles = cacheStyles(theme => ({
     // Layout:
     minHeight: theme.rem(2.75),
     padding: theme.rem(0.75),
-    backgroundColor: theme.settingsHeaderRowBackground,
```
Comment (amanalili) [inline src/components/common/SettingsHeaderRow.js]: retain settingsHeaderRowBackground

File: src/theme/variables/edgeDark.js [hunk 15]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
+  glassButtonDarkIcon: palette.white,
+
+  cardBackground: palette.edgeBlue,
+  cardShadow: palette.blackOp25,
+
+  tabBarBackground: palette.edgeNavy,
+  tabBarIcon: palette.white,
+  tabBarIconHighlighted: palette.edgeMint,
+
+  sendTab: palette.accentRed, // sliderTabSend
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:116]: sliderTabSend, sliderTabRequest, sliderTabMore

File: src/theme/variables/edgeDark.js [hunk 16]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
+  confirmationThumbDeactivated: palette.gray,
+
+  // Lines
+  lineDivider: palette.blueGray, // change from listDivider
+  textInputLine: palette.blueGray,
+  orLine: palette.blueGray,
+  tileDivider: palette.blueGray,
+
+  // Notifications
+  notificationBackground: palette.lightGrayOp75,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:147]: What is this for

File: src/theme/variables/edgeDark.js [hunk 17]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
+  messageBanner: palette.grayOp80,
+
+  // Alert Modal
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:150]: rename all to securityAlertModal

File: src/theme/variables/edgeDark.js [hunk 18]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
+  alertModalWarningIcon: palette.accentOrange,
+  alertModalDangerIcon: palette.accentRed,
+  alertModalBackground: palette.white,
+  alertModalText: palette.black,
+  alertModalLine: palette.lightGray,
+  alertModalHeaderIconShadow: palette.accentOrangeOp30,
+
+  // Settings Row
+  settingsRowBackground: palette.edgeBlue,
+  settingsRowUnderlayColor: palette.transparent,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:162]: settingsRowPressed

File: src/theme/variables/edgeDark.js [hunk 19]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
+  dangerText: palette.accentRed,
+  textLink: palette.edgeMint,
+  deactivatedText: palette.gray,
+
+  // Header
+  headerText: palette.white,
+  hamburgerButton: palette.white,
+  backButton: palette.white,
+
+  // Buttons
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:90]: Should have a pressed color

File: src/theme/variables/edgeDark.js [hunk 20]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
-  datetimepickerBackgroundDefault: pallete.white,
-  datetimepickerBackgroundDark: pallete.edgeBlue,
+  // Remove defaults when lightMode is implemented
+  keyboardTopViewBackgroundDefault: palette.white,
+  keyboardTopViewBackgroundDark: palette.edgeBlue,
+  keyboardTopViewTextDefault: palette.accentBlue,
+  keyboardTopViewTextDark: palette.white,
+  datetimepickerBackgroundDefault: palette.white,
+  datetimepickerBackgroundDark: palette.edgeBlue,
+  pressedOpacity: 0.25,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:173]: Should be removed when new pressed colors are given

File: src/theme/variables/edgeDark.js [hunk 21]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
-  secondaryButtonOutline: pallete.edgeMint,
-  secondaryButtonText: pallete.edgeMint,
-  selectButtonOutline: pallete.edgeMint,
-  selectButtonText: pallete.edgeMint,
-  modalBody: pallete.edgeNavy,
-  modalClose: pallete.white,
-  modalBackgroundShadow: pallete.black,
+  // Icons
+  icon: palette.white,
+  iconHighlighted: palette.edgeMint,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:54]: rename to iconTappable

File: src/theme/variables/edgeDark.js [hunk 22]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
+  tileIcon: palette.white,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:55]: merge To use icon

File: src/theme/variables/edgeDark.js (hunk 12 above)
Comment (amanalili) [reply src/theme/variables/edgeDark.js:57]: remove closeIcon

File: src/theme/variables/edgeDark.js [hunk 23]
```diff
@@ -14,88 +14,169 @@ import totleLogo from '../../assets/images/exchange/settingsExchangeTotle.png'
...
-  tileBackground: pallete.edgeBlue,
-  tileIcon: pallete.edgeMint,
-  tileMore: pallete.white,
-  lineDivider: pallete.edgeBlue,
-  background1: pallete.darkBlueNavyGradient1,
-  background2: pallete.darkBlueNavyGradient2,
-  cancelButton: pallete.white,
-  accentTextPositive: pallete.accentGreen,
-  accentTextNegative: pallete.accentRed,
+  modalCloseIcon: palette.white,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:68]: use edgeMint

File: src/components/scenes/SwapSettingsScene.js [hunk 24]
```diff
@@ -181,7 +181,7 @@ export class SwapSettings extends Component<Props, State> {
 
 const getStyles = cacheStyles(theme => ({
   instructionArea: {
-    backgroundColor: theme.settingsHeaderRowBackground,
+    backgroundColor: theme.listHeaderBackground,
```
Comment (amanalili) [inline src/components/scenes/SwapSettingsScene.js:183]: should use settingsRowSubHeader = palette.transparent. Ask designer what design should be.

File: src/components/scenes/TransactionDetailsScene.js [hunk 25]
```diff
@@ -556,7 +556,7 @@ const getStyles = cacheStyles(theme => ({
     color: theme.primaryText
   },
   textTransactionData: {
-    color: theme.selectButtonText,
+    color: theme.secondaryButtonText,
```
Comment (amanalili) [inline src/components/scenes/TransactionDetailsScene.js:559]: rename to textLink

File: src/components/scenes/TransactionDetailsScene.js [hunk 26]
```diff
@@ -556,7 +556,7 @@ const getStyles = cacheStyles(theme => ({
     color: theme.primaryText
   },
   textTransactionData: {
```
Comment (amanalili) [inline src/components/scenes/TransactionDetailsScene.js:558]: textAdvancedTransaction

File: src/components/scenes/NotificationScene.js [hunk 27]
```diff
@@ -1,7 +1,7 @@
 // @flow
 
 import { type EdgeCurrencyInfo } from 'edge-core-js'
-import * as React from 'react'
+import React, { Component } from 'react'
```
Comment (swansontec) [inline src/components/scenes/NotificationScene.js:4]: We just did a giant refactor to switch everything over to `import * as React from 'react'`, so please leave this the way it was. The * import is the "correct" way to use React going forward, according to tweets from the React team. Also, it's the only way that works out-of-the-box in Typescript without special config tweaks.

File: src/theme/variables/edgeDark.js (hunk 16 above)
Comment (amanalili) [reply src/theme/variables/edgeDark.js:145]: Comments everything not used and notify designer what are they used for

File: src/theme/variables/edgeDark.js [hunk 28]
```diff
@@ -1,59 +1,189 @@
...
+
+  // Settings Row
+  settingsRowBackground: palette.edgeBlue,
+  settingsRowPressed: palette.transparent,
+  settingsRowHeaderBackground: palette.edgeNavy,
+  settingsRowSubHeader: palette.transparent,
+
+  // Misc
+  // Remove defaults when lightMode is implemented
+  keyboardTopViewBackgroundDefault: palette.white,
```
Comment (amanalili) [inline src/theme/variables/edgeDark.js:166]: Should be keyboardTopViewBackgroundLight

**Pull request merged**